rust/kernel-r/src/lib.rs

```diff
@@ -0,0 +1,835 @@
+use std::sync::atomic::{AtomicU64, Ordering};
+
+use kernel_micro::{
+    common::eyre::Result, format::Format, Kernel, KernelAvailability, KernelForks, KernelInstance,
+    KernelInterrupt, KernelKill, KernelTerminate, Microkernel,
+};
+
+/// A kernel for executing R code
+#[derive(Default)]
+pub struct RKernel {
+    /// A counter of instances of this microkernel
+    instances: AtomicU64,
+}
+
+impl Kernel for RKernel {
+    fn id(&self) -> String {
+        "R".to_string()
+    }
+
+    fn availability(&self) -> KernelAvailability {
+        self.microkernel_availability()
+    }
+
+    fn supports_languages(&self) -> Vec<Format> {
+        vec![Format::R]
+    }
+
+    fn supports_interrupt(&self) -> KernelInterrupt {
+        self.microkernel_supports_interrupt()
+    }
+
+    fn supports_terminate(&self) -> KernelTerminate {
+        self.microkernel_supports_terminate()
+    }
+
+    fn supports_kill(&self) -> KernelKill {
+        self.microkernel_supports_kill()
+    }
+
+    fn supports_forks(&self) -> KernelForks {
+        // Uses R `os.fork()` which is only available on POSIX-based systems
+        // TODO: fix for R
```

Comment:
See the implementation of `microkernel_supports_forks` suggest this change:

```suggestion
        // not Windows so use `microkernel_supports_forks()`
```